documentation/Database-Schema-Application.md

```diff
@@ -0,0 +1,56 @@
+The `template_id` links to the template used to display all sections and elements (questions) to the user.
+The `user_id` links to the Applicant user that created this application.
+The `serial` is a number used to display the application number to users. 
+The `name` will be deduced by the template name + user + company names.
+The `is_active` is set to `'true'` for all applications that are still in the review process. 
+The `is_active` is set to `'false'` for applications that have finished the review process, have expire or were withdrawn by the Applicant. 
+The `trigger` is updated everytime the application has changes done by users or a scheduler. The trigger would be one of the options: `'onApplicationCreate'`, `'onApplicationSubmit'`, `'onApplicationSave'`, `'onApplicationWithdrawn'`, `'onReviewStart'`, `'onReviewEditComment'`, `'onReviewSave'`, `'onReviewAssign'`, `'onApprovalSubmit'`, `'onScheduleTime'`.
+The `outcome` is either `'Pending'`, `'Approved'`, `'Rejected'`. The application is 'Pending' during the review process, each stage will store also the outcome, so it will only be updated to here once the review process is finished resulting in either 'Approved' or 'Rejected'.
+<b>To be considered:</b>
+   * The `unique_identifier` is a new field. What is it used for? Should is replace the `serial`?
+   * An application with Draft should be flagged as `is_active`?
+
+* <b>application_section</b>
+Sections of the application that contain some elements. This is exactly the same as the `template_section`.
+The `application_id` links to the application.
+The `template_section_id` links to the template.
+<b>To be considered: Why so we need an `application_section` and not `application_element`. We should use the same logic for both.</b>
+
+* <b>application_stage_history</b>
+There is one or more stages per application template. Each one is defined by `template_stage`. The application stage history will keep records for when the application changed to each particular stage (previously defined for this application template).
+The `application_id` links to the application.
+The `stage_id` links to the template stage, which will have the `name` and number of this stage.
+The `time_created` defined when this stage was created.
+The `is_current` is set to `'true'` while this is the current stage of this application. Otherwise is set to `'false'`.
+<b>To be considered: When the application review is not in progress anymore do we keep the last `is_active` stage as `'true'`?</b>
+
+* <b>application_status_history</b>
+For each stage of the application it can have associated statuses. The application status history will keep records for when the application changed to each particular status of a stage. 
+The `application_stage_history_id` links to the application stage (and the application itself).
+The `status` is either `'Draft'`, `'Withdrawn'`, `'Submitted'`, `'Changes Required'`, `'Re-submitted'`, `'Completed'` and self-explanatory,
```

Comment:
I think you can lose the "and self-explanatory" from the end. If it's self-explanatory (which it is), why explain that it's self-explanatory? Just let it explain itself -- it's self-explanatory after all! 🤣

Reply:
hahahhaha. Yes. I agree. Don't need to explain the inexplainable.